### PR TITLE
Added support for CentOS 7 / MySQL 8

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: irods_testing_environment/odbc_setup.py [hunk 1]
```diff
@@ -238,6 +238,22 @@ def configure_odbc_driver_centos_7_mysql_57(csp_container, odbc_driver):
...
+    """Configure ODBC driver for mysql 8.0 on centos 7.
+
+    Argument:
+    csp_container -- docker container on which the iRODS catalog service provider is running
+    odbc_driver -- path to local archive file containing the ODBC driver package
+    """
+    if not odbc_driver:
+        odbc_driver = download_mysql_odbc_driver(
+            #'https://dev.mysql.com/get/Downloads/Connector-ODBC/8.0/mysql-connector-odbc-8.0.31-1.el7.x86_64.rpm')
+            #'https://dev.mysql.com/get/Downloads/Connector-ODBC/8.0/mysql-connector-odbc-8.0.31-linux-glibc2.27-x86-64bit.tar.gz')
```
Comment (alanking) [inline irods_testing_environment/odbc_setup.py:251]: Any reason to keep these around? It's fine with me, just curious. Maybe worth a comment.

**Review by alanking — COMMENTED** [review]
Looks good to me. Please create/find an issue number and I think this is good to go

File: irods_testing_environment/odbc_setup.py (hunk 1 above)
Comment (korydraughn) [reply irods_testing_environment/odbc_setup.py:251]: I don't think we need them. They are leftovers for when I was trying to make it work.

At some point, we'll need to bump the version of the database. We can revisit the drivers when that time comes.

File: projects/centos-7/centos-7-mysql-8.0/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        #
+        # --disable-log-bin is needed so that setup succeeds. Without this option, the following
+        # error is produced and setup fails:
+        #
+        #    ERROR 1418 (HY000): This function has none of DETERMINISTIC, NO SQL,
+        #    or READS SQL DATA in its declaration and binary logging is enabled
+        #    (you *might* want to use the less safe log_bin_trust_function_creators
+        #    variable)
+        #
+        command: "--transaction-isolation=READ-COMMITTED --collation-server=utf8mb4_0900_as_cs --disable-log-bin --default-authentication-plugin=mysql_native_password"
```
Comment (korydraughn) [inline projects/centos-7/centos-7-mysql-8.0/docker-compose.yml:23]: Probably need to verify if the final parameter is actually needed.

File: irods_testing_environment/odbc_setup.py (hunk 1 above)
Comment (korydraughn) [reply irods_testing_environment/odbc_setup.py]: Removed the lines.

File: projects/centos-7/centos-7-mysql-8.0/docker-compose.yml (hunk 2 above)
Comment (korydraughn) [reply projects/centos-7/centos-7-mysql-8.0/docker-compose.yml:23]: The final option is required. Without it, setup fails with:
```bash
ERROR 2059 (HY000): Authentication plugin 'caching_sha2_password' cannot be loaded: /usr/lib64/mysql/plugin/caching_sha2_password.so: cannot open shared object file: No such file or directory
```

Comment (korydraughn) [reply projects/centos-7/centos-7-mysql-8.0/docker-compose.yml:23]: Created issue #151 for investigation.

**Review by alanking — APPROVED** [review]
\# it up

**Comment by korydraughn** [comment]
Added #

**Pull request merged**